### DISCUSSION TITLE
How can I get more documentations on using and configuring karavan?

### DISCUSSION THREAD

**Discussion by trickMin** [apache/camel-karavan#1475] [Q&A]
Karavan looks like a very interesting and powerful integration platform. I found the abundant documentations of camel-core on the official website, but did not see the relevant documentation of karavan. Is there any corresponding usage and component configuration guidance at present? What about the content planning of subsequent documents?

**Comment by davsclaus** [comment]
What kind of documentation are you looking for?

A goal of karavan is to make the UI designer intuitive so users can start learning to use it by reading a little bit of docs, and then try some examples, play on their own etc. And watch existing videos etc.

Apache Camel in general is a large framework, and if you need to go deeper, then it takes time to master. For that the documentation on the Camel website can help, or for some users then buying one of the recent Camel books makes them get a lot more out of Camel than users whom dont do that.

ASF is an open community so help is very welcome.
https://camel.apache.org/community/contributing/

If you get more experienced with Karavan, then what others have done in the past, is to write their own blog post about their journey and share this with the internet. Its also possible to have a blog post published on the Camel website.
https://github.com/apache/camel-website

However back to your question. Can you elaborate a bit more what specific documentation you are looking for, with Karavan

**Comment by trickMin** [comment]
I have browsed the content on the Camel docs website (https://camel.apache.org/docs/) and found very detailed documentation for Camel-core and Camel-k. However, for Karavan, I only found the introduction documentation on the GitHub README.md.


I am sure that Karavan provides a wealth of visual components, but I do not know how to build a complete demo with them, such as configuring the runtime environment or exporting a usable program. I probably need a guide to help me get started with Karavan quickly. I think I can study more advanced and complex functions on my own.

I try runtime for Karavan, but VS code prompt "no application", How can I configure it?
![image](https://user-images.githubusercontent.com/59214333/234156826-b69a44c1-0b44-4b80-9062-e1b2e52119b1.png)

There are three default modules in VS Code. What is the relation between RestAPI and Routes? If I configure Beans, where should I put the Bean class?
![image](https://user-images.githubusercontent.com/59214333/234156971-d4dee086-a445-4298-aab3-fc0a06cf332b.png)

These are just a few examples I have cited, and there may be more. I have not found any relevant documentation to guide me on these issues. Therefore, I hope there is a step-by-step guide to help me use a complete integrated application with Karavan smoothly

**Comment by cdia8890** [comment]
@mgubaidullin is it possible to add a small explanation on how to configure and use Beans using Karavan UI? This will be a great help.

Thanks.
